plugins/orchestrator/src/components/NewWorkflowViewerPage/NewWorkflowViewerPage.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { useCallback } from 'react';
+import { useForm, UseFormRegisterReturn } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
+
+import { InfoCard } from '@backstage/core-components';
+import { errorApiRef, useApi, useRouteRef } from '@backstage/core-plugin-api';
+
+import {
+  Box,
+  Button,
+  Divider,
+  FormControl,
+  Grid,
+  TextField,
+  Typography,
+  useMediaQuery,
+  useTheme,
+} from '@material-ui/core';
+
+import {
+  workflow_json_sample,
+  workflow_yaml_sample,
+  WorkflowFormat,
+} from '@janus-idp/backstage-plugin-orchestrator-common';
+
+import { orchestratorApiRef } from '../../api';
+import {
+  createWorkflowRouteRef,
+  workflowDefinitionsRouteRef,
+} from '../../routes';
+import { BaseOrchestratorPage } from '../BaseOrchestratorPage/BaseOrchestratorPage';
+
+type FormData = {
+  url: string;
+};
+
+export const NewWorkflowViewerPage = () => {
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
+  const errorApi = useApi(errorApiRef);
+  const orchestratorApi = useApi(orchestratorApiRef);
+  const createWorkflowLink = useRouteRef(createWorkflowRouteRef);
+
+  const defaultValues: FormData = {
+    url: workflow_json_sample.url,
+  };
+  const { handleSubmit, register, formState } = useForm<FormData>({
+    defaultValues,
+    mode: 'onChange',
+  });
+
+  const { errors } = formState;
+
+  const navigate = useNavigate();
+  const definitionLink = useRouteRef(workflowDefinitionsRouteRef);
+
+  const handleResult = useCallback(
+    async ({ url }: FormData) => {
+      if (!url) {
+        return;
+      }
+      try {
+        const result = await orchestratorApi.createWorkflowDefinition(url);
+
+        if (!result?.definition.id) {
+          errorApi.post(new Error('error importing workflow'));
+        } else {
+          const workflowFormat = result.uri.endsWith('.json') ? 'json' : 'yaml';
+          navigate(
+            definitionLink({
+              workflowId: result.definition.id,
+              format: workflowFormat,
+            }),
+          );
+        }
+      } catch (e: any) {
+        errorApi.post(new Error(e));
+      }
+    },
+    [orchestratorApi, errorApi, navigate, definitionLink],
+  );
+
+  const newWorkflow = useCallback(
+    (format: WorkflowFormat) => {
+      navigate(
+        createWorkflowLink({
+          format,
+        }),
+      );
+    },
+    [createWorkflowLink, navigate],
+  );
+
+  function asInputRef(renderResult: UseFormRegisterReturn) {
+    const { ref, ...rest } = renderResult;
+    return {
+      inputRef: ref,
+      ...rest,
+    };
+  }
+
+  const contentItems = [
+    <Grid item xs={12} xl={6} key="create-workflow">
+      <InfoCard title="Create" subheader="Start authoring from scratch">
+        <Button
+          color="default"
+          variant="outlined"
+          style={{ marginTop: 8, marginRight: 8 }}
+          onClick={() => newWorkflow('yaml')}
```

Comment:
By doing this, the effect of `useCallback` is gone - there will be a new function at every render